.gitignore

```diff
@@ -1,4 +1,456 @@
-.vscode/
-*.pyc
-/build/
-*~
\ No newline at end of file
+# Created by https://www.toptal.com/developers/gitignore/api/python,c,c++,pycharm,visualstudiocode,vim,emacs,linux,windows,macos
```

Comment:
please revert changes in this file, although adding `.DS_Store` in here would be helpful, but no need to put this large auto-generated one

Reply:
+1